src/Fist/Http/tests/RequestTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+use Fist\Http\Request;
+use Fist\Testing\TestCase;
+
+class RequestTest extends TestCase
+{
+    public function testBuildingInstance()
+    {
+        $request = new Request();
+
+        $this->assertInstanceOf(Request::class, $request);
+    }
+
+    public function testBuildingFromGlobals()
+    {
+        $request = Request::createFromGlobals();
+
+        $this->assertInstanceOf(Request::class, $request);
+    }
+
+    public function testGettingRequestMethod()
+    {
+        $request = Request::createFromGlobals();
+
+        $this->assertEquals('GET', $request->getMethod());
+    }
+
+    public function testGettingFakedRequestMethod()
+    {
+        $_SERVER['REQUEST_METHOD'] = 'POST';
+
+        $request = Request::createFromGlobals();
+        unset($_SERVER['REQUEST_METHOD']);
+
+        $this->assertEquals('POST', $request->getMethod());
+    }
+
+    public function testOverwritingRequestMethod()
+    {
+        $request = Request::createFromGlobals();
+
+        $this->assertEquals('GET', $request->getMethod());
+
+        $request->setMethod('PATCH');
+
+        $this->assertEquals('PATCH', $request->getMethod());
+    }
+
+    public function testGettingRequestUrl()
+    {
+        $_SERVER['SERVER_NAME'] = 'foo.bar';
+        $_SERVER['HTTP_X_ORIGINAL_URL'] = '/baz';
+
+        $request = Request::createFromGlobals();
+        unset($_SERVER['SERVER_NAME']);
+        unset($_SERVER['HTTP_X_ORIGINAL_URL']);
+
+        $this->assertEquals('http://foo.bar/baz', $request->getUrl());
+    }
+
+    public function testGettingRequestUrlWithQuery()
+    {
+        $_SERVER['SERVER_NAME'] = 'foo.bar';
+        $_SERVER['HTTP_X_ORIGINAL_URL'] = '/baz';
+        $_SERVER['QUERY_STRING'] = 'foo=bar&bar=baz';
+
+        $request = Request::createFromGlobals();
+        unset($_SERVER['SERVER_NAME']);
+        unset($_SERVER['HTTP_X_ORIGINAL_URL']);
+        unset($_SERVER['QUERY_STRING']);
+
+        $this->assertEquals('http://foo.bar/baz?foo=bar&bar=baz', $request->getUrl());
+    }
+
+    public function testGettingRequestPath()
+    {
+        $_SERVER['SERVER_NAME'] = 'foo.bar';
+        $_SERVER['HTTP_X_ORIGINAL_URL'] = '/baz';
+        $_SERVER['QUERY_STRING'] = 'foo=bar&bar=baz';
+
+        $request = Request::createFromGlobals();
+        unset($_SERVER['SERVER_NAME']);
+        unset($_SERVER['HTTP_X_ORIGINAL_URL']);
+        unset($_SERVER['QUERY_STRING']);
+
+        $this->assertEquals('/baz', $request->getPath());
+    }
+
+    public function testGettingRequestSegments()
```

Comment:
__[phpmd]__
* testGettingRequestSegments accesses the super-global variable $_SERVER.

<!-- Comment by [SideCI](https://sideci.com) -->